File: SmashSonic/Models/ServerConfig.swift
import Foundation

struct ServerConfig: Codable, Equatable {
    var serverURL: String
    var username: String
    var password: String

    var isConfigured: Bool {
        !serverURL.isEmpty && !username.isEmpty && !password.isEmpty
    }

    var baseURL: URL? {
        var urlString = serverURL.trimmingCharacters(in: .whitespacesAndNewlines)
        if !urlString.hasPrefix("http://") && !urlString.hasPrefix("https://") {
            urlString = "https://" + urlString
        }
        if urlString.hasSuffix("/") {
            urlString = String(urlString.dropLast())
        }
        return URL(string: urlString)
    }

    static let empty = ServerConfig(serverURL: "", username: "", password: "")
}

File: SmashSonic-Android/app/src/main/java/com/smashsonic/data/model/ServerConfig.kt
package com.smashsonic.data.model

data class ServerConfig(
    val serverURL: String = "",
    val username: String = "",
    val password: String = "",
) {
    val isConfigured: Boolean
        get() = serverURL.isNotBlank() && username.isNotBlank() && password.isNotBlank()

    val baseURL: String
        get() {
            var url = serverURL.trim()
            if (!url.startsWith("http://") && !url.startsWith("https://")) {
                url = "https://$url"
            }
            return url.trimEnd('/')
        }

    companion object {
        val Empty = ServerConfig()
    }
}
